src/parser/validator.cc

```diff
@@ -82,130 +88,122 @@ void Validator::ValidateLaneAdjacency(const Parser* parser, const ValidatorConfi
         const Lane& lane = lanes[idx];
 
         // Note: The following code is a bit repetitive for left and right adjacency checks, but it is easier to read
-        // and understand. Check right adjacency <--------------------------------------------------
+        // and understand.
+        //
+        // Check right adjacency <--------------------------------------------------
         if (lane.right_lane_id) {
           // Check if there is a idx to the right.
-          if (idx == 0) {
-            // Error.
-            const std::string msg{"Wrong ordering of lanes: Lane " + lane.id + " has a right lane id (" +
-                                  lane.right_lane_id.value() + ") but is the first lane in the segment " +
-                                  segment.first + "."};
-            Report(msg, Validator::Error::Type::kLaneAdjacency, Validator::Error::Severity::kError);
-          }
+          evaluate(idx == 0,
+                   {"Wrong ordering of lanes: Lane " + lane.id + " has a right lane id (" + lane.right_lane_id.value() +
+                    ") but is the first lane in the segment " + segment.first + "."},
+                   Validator::Error::Type::kLogicalLaneAdjacency);
           // Check if right lane id is in the same segment
           const auto adj_lane_it = std::find_if(lanes.begin(), lanes.end(), [&lane](const Lane& lane_it) {
             return lane.right_lane_id.value() == lane_it.id;
           });
           if (adj_lane_it == lanes.end()) {
             // Error.
-            const std::string msg{"Adjacent lane isn't part of the segment: Lane " + lane.id +
-                                  " has a right lane id (" + lane.right_lane_id.value() +
-                                  ") that is not part of the segment " + segment.first + "."};
-            Report(msg, Validator::Error::Type::kLaneAdjacency, Validator::Error::Severity::kError);
+            evaluate(true,
+                     {"Adjacent lane isn't part of the segment: Lane " + lane.id + " has a right lane id (" +
+                      lane.right_lane_id.value() + ") that is not part of the segment " + segment.first + "."},
+                     Validator::Error::Type::kLogicalLaneAdjacency);
           } else {
             // Check if right lane id has the lane id as left lane id.
             if (adj_lane_it->left_lane_id) {
```

Comment:
Same comment here.

Reply:
Actually, if you think it'll be clearer, you can remove the if-else logic and place the condition evaluation inplace with the full path and a sequence of `evaluate` statements.

Reply:
It is true. Done.